test/integration/create_test.js

```diff
@@ -153,4 +153,40 @@ vows.describe('create').addBatch({
       }
     },
   }
+}).addBatch({
+  'with sql-file and a bad migration, causes an exit' : {
+    topic: function() {
+      var configOption = path.join("--sql-file");
+      wipeMigrations(function(err) {
+        assert.isNull(err);
+        dbMigrate('create', 'sixth migration', configOption).on('exit', function() {
+          var files = fs.readdirSync(path.join(__dirname, 'migrations'));
+
+          for (var i = 0; i<files.length; i++) {
+            var file = files[i];
+            var stats = fs.statSync(path.join(__dirname, 'migrations', file));
+            if (stats.isFile() && file.match(/sixth-migration\.js$/)) {
+              fs.writeFileSync(path.join(__dirname, 'migrations', file), 'asdfghij;');
+              dbMigrate('up').on('exit', this.callback);
+            }
+          }
+        }.bind(this));
+      }.bind(this));
+    },
+    'does cause an error': function(error, code) {
+      assert.equal(error, 1);
```

Comment:
Due to promises errors now are on stderr instead of stdout. So if listen on the exit event, it returns null on failure.